docs/oppen.md

```diff
@@ -0,0 +1,82 @@
+# Derek C. Oppen's pretty pritning algorithm
+
+Derek C. Oppen's pretty printing algorithm is a method for formatting code to make it more readable and visually appealing. It arranges code with indentation and line breaks. Oppen's algorithm has influenced many tools for code formatting and remains an important technique in software development for improving code clarity.
+
+## The printing tokens
+
+### String
+
+...?
+
+### Begin
+
+...?
+
+### End
+
+...?
+
+### Break
+
+...?
+
+### LineBreak
+
+...?
+
```

Comment:
No need for many subheading. A table of `(id, description)` suffices (you can use HTML or if your editor works nice with md tables, then why not).